test/lib/templateFactory.test.js

```diff
@@ -1022,6 +1037,53 @@ describe('Template Factory', () => {
                 }
             ];
 
+            pipelineJobs = [
+                {
+                    templateId: 1,
+                    pipelineId: 4
```

Comment:
Include `jobId` for all the entries so that we don't confused on why we have duplicate entries of `templateId` and `pipelineId` pair.
```suggestion
                    templateId: 1,
                    pipelineId: 4,
                    jobId: 41
```